### PR TITLE
Adding topics to df using openAI

### PR DESCRIPTION
**Pull request by megansam5**
solves #8 
Adds a topics column to the dataframe with the associated topics, removes rows with no topics.

### CODE REVIEW

File: news_sentiment_analyser/openai_topics.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        yield lst[i:i + chunk_size]
+
+
+def find_article_topics(article_titles: list[str], topics: list[str], openai_client: OpenAI) -> str:
+    """Returns a dictionary of article title to topics associated with each article title."""
+    titles = "\n".join(article_titles)
+    system_content = create_message(topics)
+    response = openai_client.chat.completions.create(
+        messages=[
+
```
Comment (YakMan101) [inline news_sentiment_analyser/openai_topics.py:45]: empty white line

File: news_sentiment_analyser/openai_topics.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    ai = OpenAI(api_key=ENV["OPENAI_API_KEY"])
+    titles = articles['title'].tolist()
+    topic_test = get_topic_names()
+    combined_topic_dict = {}
+    # This would be in batches of 3, probably increase to 10/20? need to test speed
+    for batch in chunk_list(titles, 3):
+        batch_topic_dict = find_article_topics(batch, topic_test, ai)
+        # Creates a dictionary of title to topics
+        combined_topic_dict.update(batch_topic_dict)
+    # Adds a column with the list of topics
```
Comment (YakMan101) [inline news_sentiment_analyser/openai_topics.py:24]: I think you can remove the comments here.

File: news_sentiment_analyser/database_functions.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+from dotenv import load_dotenv
+
+
+def create_connection() -> connection:
+    """Creates a connection to the RDS with postgres."""
+    load_dotenv()
+    conn = connect(dbname=ENV["DB_NAME"], user=ENV["DB_USER"],
+                   host=ENV["DB_HOST"], password=ENV["DB_PASSWORD"],
+                   port=ENV["DB_PORT"],
+                   cursor_factory=RealDictCursor)
```
Comment (YakMan101) [inline news_sentiment_analyser/database_functions.py:16]: Nice good old RealDictCursor makes response easier.

File: news_sentiment_analyser/database_functions.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    return conn
+
+
+def get_topic_names() -> list[str]:
+    """Returns a list of topic names."""
+    conn = create_connection()
```
Comment (YakMan101) [inline news_sentiment_analyser/database_functions.py:23]: Use with `create_connection() as conn:` for safe exits incase of error

**Review by indiah444 — APPROVED** [review]
All functions look good, clear docstrings and good idea to separate the database functions from the OpenAI functions.

**Pull request merged**